contracts/RMRK/extension/tokenProperties/IRMRKPropertiesAccessControl.sol

```diff
@@ -0,0 +1,157 @@
+        address[] collaboratorAddresses,
+        bool[] collaboratorAddressAccess,
+        address[] specificAddresses,
+        bool[] specificAddressAccess
+    );
+
+    /**
+     * @notice Used to register a collection to use the RMRK token properties repository.
+     * @dev  If the collection does not implement the Ownable interface, the `useOwnable` value must be set to `false`.
+     * @param collection The address of the collection that will use the RMRK token properties repository.
+     * @param issuer The address of the issuer of the collection.
+     * @param useOwnable The boolean value to indicate if the collection implements the Ownable interface and whether it
+     *  should be used to validate that the caller is the issuer (`true`) or to use the manually set issuer address
+     *  (`false`).
+     */
+    function registerAccessControl(
+        address collection,
+        address issuer,
+        bool useOwnable
+    ) external;
+
+    /**
+     * @notice Used to manage the access control settings for a specific parameter.
+     * @dev Only the `issuer` of the collection can call this function.
+     * @dev The `collaboratorAddresses` and `collaboratorAddressAccess` arrays must be of the same length.
+     * @dev The `specificAddresses` and `specificAddressAccess` arrays must be of the same length.
+     * @dev the possible `parameterType` values are:
+     *  [
+     *      STRING,
+     *      ADDRESS,
+     *      BYTES,
+     *      UINT,
+     *      BOOL
+     *  ]
+     * @dev The possible `accessType` values are:
+     *  [
+     *      Issuer,
+     *      Collaborator,
+     *      IssuerOrCollaborator,
+     *      TokenOwner,
+     *      SpecificAddress,
+     *  ]
+     * @dev Emits an {AccessControlUpdated} event.
+     * @param collection The address of the collection being managed.
+     * @param parameterType Type of the parameter being managed.
+     * @param parameterId ID of the parameter being managed.
+     * @param accessType The type of access control to be applied to the parameter.
+     * @param collaboratorAddresses The array of addresses to be added or removed from the list of collaborators.
+     * @param collaboratorAddressAccess The array of boolean values to indicate if the address should be added as a
+     *  collaborator (`true`) or removed (`false`).
+     * @param specificAddresses The array of addresses to be added or removed from the list of specific addresses.
+     * @param specificAddressAccess The array of boolean values to indicate if the address should be added as a specific
+     *  address (`true`) or removed (`false`).
+     */
+    function manageAccessControl(
+        address collection,
+        ParameterType parameterType,
+        uint256 parameterId,
+        AccessType accessType,
+        address[] memory collaboratorAddresses,
+        bool[] memory collaboratorAddressAccess,
+        address[] memory specificAddresses,
+        bool[] memory specificAddressAccess
```

Comment:
I don't understand this interface. 
First, what's the need for the bool array? Is it to enable/disable at the same time? It's weird anyway. I expected collaborators to be set by the owner per collection, no need to set them for every property.
Regarding the specific addresses, it would be much simpler if it was just one. So no need to use 2 lists, just a single address which will replace whatever there is for the collection and parameter (which I think should be gated by name, and not id, the id should be an inner thing the users don't have to care about).
If we want to still have a list, let's have only that, no need for the boolean list. Everytime we set it it can simply replace the list, the bad thing is that it would imply a for loop when setting a property to check if the caller is in that list (for the case access type is specific address)

Reply:
Yeah, the idea was to allow to add and remove any number of collaborators at a time and leave the states of already set ones alone if needed.
Having collaborators defined on a collection level makes sense, I agree and I'll modify this. 👌 
Ok, I don't like the idea of the for loop, so I think a specific address on a per-key basis seems best. 🤔